bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyBridgeHandler.java

```diff
@@ -0,0 +1,186 @@
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * {@link LinkyBridgeHandler} is the base handler to access enedis data.
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+@NonNullByDefault
+public abstract class LinkyBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(LinkyBridgeHandler.class);
+
+    protected final HttpService httpService;
+    protected final BundleContext bundleContext;
+    protected final HttpClient httpClient;
+    protected final EnedisHttpApi enedisApi;
+    protected final ThingRegistry thingRegistry;
+
+    protected final LinkyConfiguration config;
+    protected final Gson gson;
+
+    protected boolean connected = false;
+
+    private static final int REQUEST_BUFFER_SIZE = 8000;
+
+    public LinkyBridgeHandler(Bridge bridge, final @Reference HttpClientFactory httpClientFactory,
+            final @Reference OAuthFactory oAuthFactory, final @Reference HttpService httpService,
+            final @Reference ThingRegistry thingRegistry, ComponentContext componentContext, Gson gson) {
+        super(bridge);
+
+        SslContextFactory sslContextFactory = new SslContextFactory.Client();
+        try {
+            SSLContext sslContext = SSLContext.getInstance("SSL");
+            sslContext.init(null, new TrustManager[] { TrustAllTrustManager.getInstance() }, null);
+            sslContextFactory.setSslContext(sslContext);
+        } catch (NoSuchAlgorithmException e) {
+            logger.warn("An exception occurred while requesting the SSL encryption algorithm : '{}'", e.getMessage(),
+                    e);
+        } catch (KeyManagementException e) {
+            logger.warn("An exception occurred while initialising the SSL context : '{}'", e.getMessage(), e);
+        }
+
+        this.gson = gson;
+        this.httpService = httpService;
+        this.thingRegistry = thingRegistry;
+        this.bundleContext = componentContext.getBundleContext();
+
+        this.httpClient = httpClientFactory.createHttpClient(LinkyBindingConstants.BINDING_ID, sslContextFactory);
+        this.httpClient.setFollowRedirects(false);
+        this.httpClient.setRequestBufferSize(REQUEST_BUFFER_SIZE);
+
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            logger.warn("Unable to start Jetty HttpClient {}", e.getMessage());
+        }
+
+        this.enedisApi = new EnedisHttpApi(this, gson, this.httpClient);
+
+        config = getConfigAs(LinkyConfiguration.class);
```

Comment:
I'm afraid getting the configuration in the constructor is a bit early. Better in `initialize` ? To be checked.

Reply:
I'm not sure about this one.
If i've moved it to initialize, I will have to remove the final keywork, and deal with possible null config ?

Reply:
Yes, because initialize is called when a configuration change happens so if you take it earlier you may miss changes.

Reply:
fix in next commit